pkg/services/ngalert/remote/alertmanager.go

```diff
@@ -126,30 +126,34 @@ func (am *Alertmanager) ApplyConfig(ctx context.Context, config *models.AlertCon
 	// First, execute a readiness check to make sure the remote Alertmanager is ready.
 	am.log.Debug("Start readiness check for remote Alertmanager", "url", am.url)
 	if err := am.checkReadiness(ctx); err != nil {
-		am.log.Error("unable to pass the readiness check", "err", err)
+		am.log.Error("Unable to pass the readiness check", "err", err)
 		return err
 	}
 	am.log.Debug("Completed readiness check for remote Alertmanager", "url", am.url)
 
+	// Send configuration if necessary.
 	am.log.Debug("Start configuration upload to remote Alertmanager", "url", am.url)
-	if ok := am.compareRemoteConfig(ctx, config); !ok {
+	if am.shouldSendConfig(ctx, config) {
 		err := am.mimirClient.CreateGrafanaAlertmanagerConfig(ctx, config.AlertmanagerConfiguration, config.ConfigurationHash, config.ID, config.CreatedAt, config.Default)
 		if err != nil {
 			am.log.Error("Unable to upload the configuration to the remote Alertmanager", "err", err)
-		} else {
-			am.log.Debug("Completed configuration upload to remote Alertmanager", "url", am.url)
 		}
 	}
+	am.log.Debug("Completed configuration upload to remote Alertmanager", "url", am.url)
 
+	// Send base64-encoded state if necessary.
 	am.log.Debug("Start state upload to remote Alertmanager", "url", am.url)
-	if ok := am.compareRemoteState(ctx, ""); !ok {
-		if err := am.mimirClient.CreateGrafanaAlertmanagerState(ctx, ""); err != nil {
+	state, err := am.state.GetFullState(ctx, notifier.SilencesFilename, notifier.NotificationLogFilename)
+	if err != nil {
+		return fmt.Errorf("error getting the Alertmanager's full state: %w", err)
```

Comment:
I think we should just log the error, not return it - the plan is to try to send it again on the next upload.